packages/server/src/middleware/nuxt.js

```diff
@@ -78,7 +79,6 @@ export default ({ options, nuxt, renderRoute, resources }) => async function nux
 
     // Send response
     res.setHeader('Content-Type', 'text/html; charset=utf-8')
-    res.setHeader('Accept-Ranges', 'none') // #3870
     res.setHeader('Content-Length', Buffer.byteLength(html))
     res.end(html, 'utf8')
     await nuxt.callHook('render:routeDone', url, result, context)
```

Comment:
There might be other use-cases to override headers and while `render:route` hook  _can_ do it it is more for context modification not doing response logic. Maybe we can add a new `render:beforeResponse` hook before `res.end` or simply swap L83 with L84?

Reply:
Second idea: we can use `res.getHeader` to ensure not overriding any header already set

Reply:
There was a similar discussion a year ago https://github.com/nuxt/nuxt.js/pull/4755
Maybe it is a good idea to go with the first option you mentioned @pi0

I will try to implement the required changes and see how it goes. Thanks for the reply!